lib/kredis/script_evaluator.rb

```diff
@@ -0,0 +1,17 @@
+module Kredis::ScriptEvaluator
+  class UnfrozenScriptError < StandardError; end
+
+  SCRIPT_DIGESTS = Hash.new do |hash, script|
+    raise UnfrozenScriptError unless script.frozen?
+
+    hash[script] = -Digest::SHA1.hexdigest(script)
+  end
+
+  extend self
+
+  def eval(redis, script, **kwargs)
+    redis.evalsha(SCRIPT_DIGESTS[script], **kwargs)
+  rescue Redis::CommandError
+    redis.eval(script, **kwargs)
+  end
```

Comment:
How would you suggest doing that? I agree with you about the evaluator being tailored for static scripts (that's in part why I had the `frozen?` check above that you commented on). That is how I intended for it to be used when I designed it. But it's also designed to abstract away the complexity of dealing with evaluating a script on Redis: namely optimizing the call to an `evalsha` which uses less bandwidth, and then dealing with the fact that the script may or may not exist in the redis instance's script cache, and so it may need to be loaded before it can be evaluated that way.

So I would say: yes, I agree, I thought this did that effectively already. Is your concern mainly with the `eval` naming? I was just trying match the redis vernacular, but I can see how it might be confusing and lead someone who is familiar with redis scripts into thinking that this is not handling the eval optimization/script loading for them. Maybe a less "redis-ized" term would be preferable? `run(script)`?